stages/org.osbuild.systemd.preset

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python3
+"""
+Configure Systemd services through presets.
+
+Enable or disable systemd services through presets.
+"""
+
+import sys
+
+import osbuild.api
+
+SCHEMA = r"""
+"additionalProperties": false,
+"properties": {
+  "presets": {
+    "type": "array",
+    "items": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["name", "state"],
+      "properties": {
+        "name": {"type": "string"},
+        "state": {"type": "string"}
+      }
+    },
+    "description": "Array of systemd unit names and their preset logic."
+  }
+}
+"""
+
+
+def main(tree, options):
+    presets = options.get("presets")
+
+    if presets:
+        svcs = {}
+
+        for svc in presets:
+            # verify that we don't have duplicate service names with different
+            # states
+            if svc["name"] in svcs:
+                raise RuntimeError(f"{svc['name']!r} has multiple defined states.")
+
+            svcs[svc["name"]] = svc["state"]
```

Comment:
nitpick: You could have define the `presets` property as an object, instead of an array, with service names being its properties (probably based on some pattern). This way you would:

- ensure that one can't specify one service multiple times
- not need to construct the dictionary with key mapping to service name and value mapping to its state, since you would get this dict "for free" directly from the stage options.

Reply:
While I agree I think the behavior is different. If you define a manifest with multiple same-keys then whatever the JSON parser likes is the one that ends up winning. Doing it this way you have an error (albeit, late in the process).

If our schema validator or such already invalidates such things then I'd prefer to use properties indeed.

Reply:
I didn't considered the case when someone writes a manifest by hand and defines the same properties in an object multiple times. AFAIK we do not consider this either in other stages which use the approach that I described (e.g. `org.osbuild.users` stage). I'm not sure if the JSON schema validator would catch this.